src/Faithlife.Utility/DirectoryUtility.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace Faithlife.Utility
+{
+	/// <summary>
+	/// Provides utility methods for working with directories.
+	/// </summary>
+	public static class DirectoryUtility
+	{
+		/// <summary>
+		/// Recursively finds all files matching <paramref name="searchPatterns"/> in the folders specified by
+		/// <paramref name="directoryPaths"/> and their subfolders.
+		/// </summary>
+		/// <param name="directoryPaths">The directory paths to search.</param>
+		/// <param name="searchPatterns">The search patterns to match against the names of files.</param>
+		/// <param name="cancellationToken">The cancellation token.</param>
+		/// <returns>A sequence of fully-qualified file paths.</returns>
+		public static IEnumerable<string> FindFiles(IEnumerable<string> directoryPaths, IEnumerable<string> searchPatterns, CancellationToken cancellationToken)
+		{
+			return DoFindFiles(directoryPaths, searchPatterns, DoFindFiles, Directory.GetDirectories, cancellationToken);
+		}
+
+		/// <summary>
+		/// Recursively finds all files matching <paramref name="searchPatterns"/> in the folders specified by
+		/// <paramref name="directoryPaths"/> and their subfolders specified by <paramref name="getDirectories"/>.
+		/// </summary>
+		/// <param name="directoryPaths">The directory paths to search.</param>
+		/// <param name="searchPatterns">The search patterns to match against the names of files.</param>
+		/// <param name="getDirectories">A function which takes a path to a directory and returns a list of subdirectories of that path.</param>
+		/// <param name="cancellationToken">The cancellation token.</param>
+		/// <returns>A sequence of FileInfo instances.</returns>
+		public static IEnumerable<FileInfo> FindFileInfos(IEnumerable<string> directoryPaths, IEnumerable<string> searchPatterns, Func<string, string[]> getDirectories, CancellationToken cancellationToken)
```

Comment:
Is this intentionally public? How is it meant to be used?

Reply:
It's used in several places in internal resource tools. Not general purpose enough? Same with `FindFiles`.

Reply:
After further review, these are confusing, rarely used, and predate `EnumerateFiles`. I'm go to remove them.